### ISSUE TITLE
Agent failed when tries to post configmap for new address

### ISSUE THREAD

**Issue by kornys** [EnMasseProject/enmasse#1313]
There is a test in systemtests.common.catalog.ServiceCatalogWebTest#testSendMessageUsingBindingCert.

Scenario:
1. Deploy enmasse with service catalog
2. Create standard address space using service catalog wizard
3. Create binding for console admin
4. Use binding for open and log in to console
5. Create queue address with some plan (sharded or pooled)
6. Address does not show in addresses tab in console and no configmap for queue exists

log from agent:
```
{"log":"2018-06-04T00:50:10.664Z agent info routers changed: amqp:/_topo/0/Router.qdrouterd-1906857074-lwgzv/$management\n","stream":"stderr","time":"2018-06-04T00:50:10.66484507Z"}
{"log":"2018-06-04T00:50:11.509Z agent info Router connected from Router.qdrouterd-1906857074-lwgzv\n","stream":"stderr","time":"2018-06-04T00:50:11.510611928Z"}
{"log":"2018-06-04T00:50:11.511Z agent info [Router.qdrouterd-1906857074-lwgzv] router ready\n","stream":"stderr","time":"2018-06-04T00:50:11.511790406Z"}
{"log":"2018-06-04T00:50:11.512Z agent info checking configuration of Router.qdrouterd-1906857074-lwgzv\n","stream":"stderr","time":"2018-06-04T00:50:11.512975825Z"}
{"log":"2018-06-04T00:50:11.524Z agent info Router.qdrouterd-1906857074-lwgzv not ready for connectivity check: false false\n","stream":"stderr","time":"2018-06-04T00:50:11.525450102Z"}
{"log":"2018-06-04T00:50:11.525Z agent info Router.qdrouterd-1906857074-lwgzv not ready for connectivity check: false true\n","stream":"stderr","time":"2018-06-04T00:50:11.525668907Z"}
{"log":"2018-06-04T00:50:11.527Z agent info addresses up to date on Router.qdrouterd-1906857074-lwgzv (ignoring 3 elements)\n","stream":"stderr","time":"2018-06-04T00:50:11.527302197Z"}
{"log":"2018-06-04T00:50:11.528Z agent info autolinks up to date on Router.qdrouterd-1906857074-lwgzv\n","stream":"stderr","time":"2018-06-04T00:50:11.528941024Z"}
{"log":"2018-06-04T00:50:11.531Z agent info linkroutes up to date on Router.qdrouterd-1906857074-lwgzv (ignoring 3 elements)\n","stream":"stderr","time":"2018-06-04T00:50:11.531890797Z"}
{"log":"2018-06-04T00:50:11.532Z agent info configuration of Router.qdrouterd-1906857074-lwgzv is up to date\n","stream":"stderr","time":"2018-06-04T00:50:11.532121589Z"}
{"log":"2018-06-04T00:50:11.532Z agent info [Router.qdrouterd-1906857074-lwgzv] addresses synchronized\n","stream":"stderr","time":"2018-06-04T00:50:11.532267237Z"}
{"log":"2018-06-04T00:50:11.532Z agent info checking connectivity for Router.qdrouterd-1906857074-lwgzv\n","stream":"stderr","time":"2018-06-04T00:50:11.532340498Z"}
{"log":"2018-06-04T00:50:11.532Z agent info [Router.qdrouterd-1906857074-lwgzv] fully connected\n","stream":"stderr","time":"2018-06-04T00:50:11.532958217Z"}
{"log":"2018-06-04T00:50:30.222Z agent info created session: 4027d567-cd02-1a41-8b02-1de59f55c6ca\n","stream":"stderr","time":"2018-06-04T00:50:30.223087938Z"}
{"log":"2018-06-04T00:50:31.939Z agent info redirecting to / (/?session_state=6220859d-d490-4151-92b9-cdd53cec111b)\n","stream":"stderr","time":"2018-06-04T00:50:31.939996314Z"}
{"log":"2018-06-04T00:50:32.961Z agent info Accepted incoming websocket connection\n","stream":"stderr","time":"2018-06-04T00:50:32.962260024Z"}
{"log":"2018-06-04T00:50:33.123Z agent info GET /api/v1/namespaces/enmasse-master-467-test-cert-messaging-space/configmaps?labelSelector=type%3Daddress-plan =\u003e 200 \n","stream":"stderr","time":"2018-06-04T00:50:33.123593733Z"}
{"log":"2018-06-04T00:50:36.921Z agent info creating address definition {\"address\":\"test-queue\",\"type\":\"queue\",\"plan\":\"sharded-queue\"}\n","stream":"stderr","time":"2018-06-04T00:50:36.921979735Z"}
{"log":"2018-06-04T00:50:36.939Z agent info POST /api/v1/namespaces/enmasse-master-467-test-cert-messaging-space/configmaps =\u003e 422\n","stream":"stderr","time":"2018-06-04T00:50:36.940400109Z"}
{"log":"2018-06-04T00:50:36.940Z agent info create_address request failed: Error: Failed to created address {\"address\":\"test-queue\",\"type\":\"queue\",\"plan\":\"sharded-queue\"}: 422 Unprocessable Entity\n","stream":"stderr","time":"2018-06-04T00:50:36.940855877Z"}
```

**Comment by grs** [comment]
@rgodfrey @lulf 

The issue here looks like it is related to the maximum length for names. The configmap being created is for an address named test-queue in address-space test-cert-messaging-space, giving a  qualified address that is 36 chars in length. However at present all addresses have a stringified uuid appended (37 chars). The maximum allowed name at present is 60 chars (63 kubernetes limit with 3 reserved for pod qualifier in stateful set

To continue with appending a stringified uuid, the address-space name would need to be less than 22 chars long to allow even a single char of the address to be used (which would limit the uniquely identifiable address to just over sixty.

In other words, I think the change to prepend the address-space without also placing stringent limits on the name of the address-space, is incompatible with the appending of a stringified uuid.

**Comment by kornys** [comment]
https://github.com/EnMasseProject/enmasse/pull/1314
